ocs_ci/utility/prometheus.py

```diff
@@ -414,27 +417,37 @@ def get(self, resource, payload=None):
         logger.debug(f"verify={self._cacert}")
         logger.debug(f"params={payload}")
 
-        response = requests.get(
-            self._endpoint + pattern,
-            headers=headers,
-            verify=self._cacert,
-            params=payload,
-        )
-        if "Application is not available" in response.text:
-            logger.warning(f"There was an error in response: {response.text}")
-            logger.warning("Refreshing connection")
-            self.refresh_connection()
-            if not config.ENV_DATA["platform"].lower() == "ibm_cloud":
-                logger.warning("Generating new certificate")
-                self.generate_cert()
-            logger.warning("Connection refreshed - trying the query again")
-            response = requests.get(
+        if timeout:
+            for sample_response in TimeoutIterator(
+                timeout=timeout,
+                sleep=15,
+                func=requests.get,
+                func_kwargs={
+                    "url": self._endpoint + pattern,
+                    "headers": headers,
+                    "verify": self._cacert,
+                    "params": payload,
+                },
+            ):
+                response = sample_response
+                if response.status_code == 503:
+                    logger.warning(f"There was an error in response: {response.text}")
+                    logger.warning("Refreshing connection")
+                    self.refresh_connection()
+                    if not config.ENV_DATA["platform"].lower() == "ibm_cloud":
+                        logger.warning("Generating new certificate")
+                        self.generate_cert()
+                    logger.warning("Connection refreshed")
+                else:
+                    break
+            return response
+        else:
+            return requests.get(
                 self._endpoint + pattern,
                 headers=headers,
                 verify=self._cacert,
                 params=payload,
             )
```

Comment:
I would rather moved this section into the beginning so that you can handle this case in a separate code block, and don't have to think about it later.